src/Chapter2.hs

```diff
@@ -726,8 +750,11 @@ value of the element itself
 
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
+-- Assuming Negative numbres will not be present in the Input Array
+-- If they appear, they will be ignored
+
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate arr = concat (map (\x -> (replicate x x)) arr)
```

Comment:
Great! 👍🏼 
If you want to write this function even more elegantly, you can use the standard function `concatMap` which is a combination of `concat` and `map` as the name suggests 🙂
After that, you could notice, that you would be able to eta-reduce on the last argument :)

Reply:
WIll make this change !